src/mixins/prevent-unsaved-mixin.ts

```diff
@@ -1,48 +1,57 @@
-import type { LitElement } from "lit";
+import type { LitElement, PropertyValues } from "lit";
 import type { Constructor } from "../types";
 import { isNavigationClick } from "../common/dom/is-navigation-click";
-import { navigate } from "../common/navigate";
+import { fireEvent } from "../common/dom/fire_event";
 
 export const PreventUnsavedMixin = <T extends Constructor<LitElement>>(
   superClass: T
 ) =>
   class extends superClass {
     private _handleClick = async (e: MouseEvent) => {
-      const href = isNavigationClick(e);
-
-      if (!href) {
+      // get the right target, otherwise the composedPath would return <home-assistant> in the new event
+      const target = e
+        .composedPath()
+        .find(
+          (n) => (n as HTMLElement).tagName === "HA-SVG-ICON"
```

Comment:
I don't like this one, you can click on other things than just the icon? Why not take just the first one in the composed path?